From e7144d11b9528238afab72ff064c07cb33745a30 Mon Sep 17 00:00:00 2001
From: Chris Thompson <cthomp@chromium.org>
Date: Tue, 11 May 2021 19:57:34 +0000
Subject: [PATCH] Update ssl_error_assistant code to py3

This is a mechanical update of ssl_error_assistant resource code to
move from Python2 to Python3. Tested by invoking the build target
(`ninja -c out_linux/Release make_ssl_error_assistant_protobuf`).

Bug: 1205600
Change-Id: I5387338e66a94af4f7dccfa3ab3872163faaeafe
Reviewed-on: https://chromium-review.googlesource.com/c/chromium/src/+/2885011
Reviewed-by: Mustafa Emre Acer <meacer@chromium.org>
Commit-Queue: Chris Thompson <cthomp@chromium.org>
Cr-Commit-Position: refs/heads/master@{#881666}
---
 .../ssl/ssl_error_assistant/BUILD.gn          |  3 +--
 .../ssl/ssl_error_assistant/PRESUBMIT.py      |  2 ++
 .../gen_ssl_error_assistant_proto.py          |  2 +-
 .../ssl/ssl_error_assistant/push_proto.py     | 22 +++++++++----------
 4 files changed, 15 insertions(+), 14 deletions(-)

diff --git a/src/components/resources/ssl/ssl_error_assistant/BUILD.gn b/src/components/resources/ssl/ssl_error_assistant/BUILD.gn
index 9a90fd463f..ca822984e1 100644
--- a/src/components/resources/ssl/ssl_error_assistant/BUILD.gn
+++ b/src/components/resources/ssl/ssl_error_assistant/BUILD.gn
@@ -5,8 +5,7 @@
 import("//build/config/python.gni")
 
 # Generate the binary proto form of "ssl_error_assistant" from the ascii proto.
-# TODO(crbug.com/1112471): Get this to run cleanly under Python 3.
-python2_action("make_ssl_error_assistant_protobuf") {
+action("make_ssl_error_assistant_protobuf") {
   script = "gen_ssl_error_assistant_proto.py"
 
   # The output goes in $target_gen_dir since that's where
diff --git a/src/components/resources/ssl/ssl_error_assistant/PRESUBMIT.py b/src/components/resources/ssl/ssl_error_assistant/PRESUBMIT.py
index 16a0f328f8..93b738b24c 100644
--- a/src/components/resources/ssl/ssl_error_assistant/PRESUBMIT.py
+++ b/src/components/resources/ssl/ssl_error_assistant/PRESUBMIT.py
@@ -7,6 +7,8 @@
    This is taken from chrome/browser/resources/safe_browsing/PRESUBMIT.py.
 """
 
+USE_PYTHON3 = True
+
 # TODO(meacer): Refactor and reuse shared code with
 #               chrome/browser/resources/safe_browsing/PRESUBMIT.py
 def CheckVersionUpdatedInSSLErrorAssistantProto(input_api, output_api):
diff --git a/src/components/resources/ssl/ssl_error_assistant/gen_ssl_error_assistant_proto.py b/src/components/resources/ssl/ssl_error_assistant/gen_ssl_error_assistant_proto.py
index bf37febd56..9032c7307d 100755
--- a/src/components/resources/ssl/ssl_error_assistant/gen_ssl_error_assistant_proto.py
+++ b/src/components/resources/ssl/ssl_error_assistant/gen_ssl_error_assistant_proto.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2016 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
diff --git a/src/components/resources/ssl/ssl_error_assistant/push_proto.py b/src/components/resources/ssl/ssl_error_assistant/push_proto.py
index e313194671..17ee45ac5d 100755
--- a/src/components/resources/ssl/ssl_error_assistant/push_proto.py
+++ b/src/components/resources/ssl/ssl_error_assistant/push_proto.py
@@ -1,4 +1,4 @@
-#!/usr/bin/env python
+#!/usr/bin/env python3
 # Copyright 2018 The Chromium Authors. All rights reserved.
 # Use of this source code is governed by a BSD-style license that can be
 # found in the LICENSE file.
@@ -46,10 +46,10 @@ def main():
   gn_command = ['ninja',
                 '-C', opts.dir,
                 RESOURCE_SUBDIR + ':make_ssl_error_assistant_protobuf']
-  print "Running the following"
-  print "   " + (' '.join(gn_command))
+  print("Running the following")
+  print("   " + (' '.join(gn_command)))
   if subprocess.call(gn_command):
-    print "Ninja failed."
+    print("Ninja failed.")
     return 1
 
   # Use the versioned files under the copy directory to push to the GCS bucket.
@@ -68,19 +68,19 @@ def main():
   version_dir = dirs[0]
   command = ['gsutil', 'cp', '-Rn', version_dir, DEST_BUCKET]
 
-  print '\nGoing to run the following command'
-  print '   ', ' '.join(command)
-  print '\nIn directory'
-  print '   ', copy_dir
-  print '\nWhich should push the following files'
+  print('\nGoing to run the following command')
+  print('   ', ' '.join(command))
+  print('\nIn directory')
+  print('   ', copy_dir)
+  print('\nWhich should push the following files')
   expected_files = [os.path.join(dp, f) for dp, _, fn in
                     os.walk(version_dir) for f in fn]
   for f in expected_files:
-    print '   ', f
+    print('   ', f)
 
   shall = raw_input('\nAre you sure (y/N) ').lower() == 'y'
   if not shall:
-    print 'aborting'
+    print('aborting')
     return 1
   return subprocess.call(command)
 
